### PR TITLE
Add UserPreference model, migrations and delegate

### PR DESCRIPTION
**Pull request by tiagograf-AOT**
## Status
**Ready**

## Fixes or Implements [AB#8625](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8625)
- [x] Enhancement
- [ ] Bug
- [ ] Documentation

## Description:
Adds user preference db model, migrations and db delegate.

## Testing
- [x] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

### UI Changes
 NO

### CODE REVIEW

File: Apps/WebClient/test/unit/Services.Test/UserProfileService_Test.cs [hunk 1]
```diff
@@ -268,16 +270,18 @@ public void ShouldGetPreference()
             UserPreference dbUserPreference = new UserPreference
             {
                 HdId = hdid,
-                DismissedMyNotePopover = true,
+                Key = "DismissedMyNotePopover",
```
Comment (FuriousLlama) [inline Apps/WebClient/test/unit/Services.Test/UserProfileService_Test.cs:273]: Lets rename it to tutorialPopover to make it general

File: Apps/Database/src/Models/UserPreference.cs [hunk 2]
```diff
@@ -38,8 +32,14 @@ public class UserPreference
         public string HdId { get; set; } = null!;
 
         /// <summary>
-        /// Gets or sets a value indicating whether the user dismissed the note popover.
+        /// Gets or sets the preference key.
+        /// </summary>
+        [Key]
+        public string Key { get; set; } = null!;
```
Comment (sslaws) [inline Apps/Database/src/Models/UserPreference.cs:38]: I'm not sure why we defined the key as the key and in the notion of the database I don't like the name of key (Maybe Preference and value).  I think this table should be hdid/preference as a composite key.

File: Apps/WebClient/src/Server/Models/UserPreferenceModel.cs [hunk 3]
```diff
@@ -16,17 +16,14 @@
 namespace HealthGateway.WebClient.Models
 {
     using System;
+    using System.Collections.Generic;
+    using System.Linq;
 
     /// <summary>
     /// Model that provides a user representation of an user preference database model.
     /// </summary>
     public class UserPreferenceModel
```
Comment (sslaws) [inline Apps/WebClient/src/Server/Models/UserPreferenceModel.cs:25]: Why are we creating a front end model?  I'm not seeing a lot of value.

Comment (leobcgov) [reply Apps/WebClient/src/Server/Models/UserPreferenceModel.cs:25]: @sslaws do you mean we can just expose the dbmodel which contains the Preference and Value pair in the Service & Controller right?

File: Apps/Database/src/Models/UserPreference.cs (hunk 2 above)
Comment (tiagograf-AOT) [reply Apps/Database/src/Models/UserPreference.cs]: Updated

File: Apps/WebClient/test/unit/Services.Test/UserProfileService_Test.cs (hunk 1 above)
Comment (tiagograf-AOT) [reply Apps/WebClient/test/unit/Services.Test/UserProfileService_Test.cs]: Updated

**Pull request merged**